### PR TITLE
Build: Enable long term caching for frontend assets

### LINKED ISSUE #42348: Webpack: Introduce long term browser caching
Webpack 5 has new algorithms for hashing chunks, modules and fixes issues with `[contenthash]` which allows for better persisting of assets in the browser cache between builds. We should consider using these to improve loading speed of Grafana between updates.

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
This PR enables Long Term Caching in the Webpack configs by taking advantage of deterministic chunk IDs, deterministic module IDs and using `[contenthash]` for filenames. This will result in longer term browser caching across Grafana version updates which _should_ result in faster loading and less network traffic.

**Which issue(s) this PR fixes**:

Fixes #42348

**Special notes for your reviewer**:
Note that when in dev mode the http server sets `Cache-Control: max-age=0, must-revalidate, no-cache` where as in production it uses `Cache-Control: public, max-age=31536000`.

I invested a little time in deleting some old loaders and moving the CopyUniconsPlugin out to its own file.

### CODE REVIEW

File: scripts/webpack/plugins/CopyUniconsPlugin.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const fs = require('fs-extra');
+const path = require('path');
+
+class CopyUniconsPlugin {
+  apply(compiler) {
```
Comment (leventebalogh) [inline scripts/webpack/plugins/CopyUniconsPlugin.js:5]: **Nit:** maybe we could add the type definition as a doc-string here as well?

```javascript
/**
* @param {import('webpack').Compiler} compiler
*/
```

File: scripts/webpack/plugins/HTMLWebpackCSSChunks.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+       */
+      (compilation) => {
+        HtmlWebpackPlugin.getHooks(compilation).beforeAssetTagGeneration.tapAsync(
+          'HTMLWebpackCSSChunks',
+          (data, cb) => {
+            data.assets.cssChunks = {};
+
+            for (const entryPoint of compilation.entrypoints.values()) {
+              for (const chunk of entryPoint.chunks) {
+                const cssFile = [...chunk.files].find((file) => file.endsWith('.css'));
```
Comment (leventebalogh) [inline scripts/webpack/plugins/HTMLWebpackCSSChunks.js:27]: **Question:** maybe a dumb question, but what if there are multiple chunk files ending with `.css`?

**Review by leventebalogh — APPROVED** [review]
Left a few minor comments, but generally LGTM! 👏

File: scripts/webpack/plugins/HTMLWebpackCSSChunks.js (hunk 2 above)
Comment (jackw) [reply scripts/webpack/plugins/HTMLWebpackCSSChunks.js:27]: Not a dumb question at all. This plugin is very geared towards our current usage of defining the css theme files as entry points which only ever output a single `css` file per entry (`dark`, `light`). Given we are aiming to migrate styles to emotion and remove scss from the codebase I wrote this as "dumb" as necessary to get the job done.

The alternative I considered to writing this plugin was to change the scss to output a single css file that used css custom variables for theming but felt that would introduce A LOT of changes and potentially styling bugs.

**Review by ashharrison90 — APPROVED** [review]
i think this looks good 🙌 

as an aside, i do think we should invest time into using css variables for the theming. couple of reasons:
- instant theme switching instead of having to do a full page reload
- it would make proper custom theming for consumers **very** easy. we'd just tell them to override some (or all) of the custom variables on the root with their own colours 🥳 

@jackw if you're interested i might attempt it during the next hackathon 🤓

**Comment by jackw** [comment]
> if you're interested i might attempt it during the next hackathon 🤓

@ashharrison90 let's pair on this and see what happens! 💪 🪄

**Pull request merged**